public/main.js

```diff
@@ -23,32 +33,29 @@ form.onsubmit = function(event) {
 };
 
 function createTodo(title, callback) {
-    var createRequest = new XMLHttpRequest();
-    createRequest.open("POST", "/api/todo");
-    createRequest.setRequestHeader("Content-type", "application/json");
-    createRequest.send(JSON.stringify({
-        title: title
-    }));
-    createRequest.onload = function() {
-        if (this.status === 201) {
-            callback();
-        } else {
-            error.textContent = "Failed to create item. Server returned " + this.status + " - " + this.responseText;
-        }
-    };
+    fetch("/api/todo/", {
```

Comment:
This is a bit incorrect, with promises, you don't need to pass in a callback, instead it should be 
```
function createTodo(title) {
  return fetch(...)
    .then(...)
    .catch(...);
}
```
And the function using it should continue using the promise, e.g. 
``` 
createTodo("").then(....);
```